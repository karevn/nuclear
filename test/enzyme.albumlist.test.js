import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { describe, it } from 'mocha';
import sinon from  'sinon';
import { Loader } from 'semantic-ui-react';

import Card from '../app/components/Card'
import AlbumList from '../app/components/AlbumList'

describe('<AlbumList />', () => {
  const mockedAlbums = [
    {
      id: "ID1",
      type: "TYPE1",
      title: 'TITLE1',
      thumb: 'THUMB1'
    },
    {
      id: "ID2",
      type: "TYPE2",
      title: 'TITLE2',
      thumb: 'THUMB2'
    },
    {
      id: "ID3",
      type: "TYPE3",
      title: 'TITLE3',
      thumb: 'THUMB3'
    }
  ];

  it('Should render loader when there is no album', () => {
    const wrapper = shallow(<AlbumList />)

    expect(wrapper.exists(Loader)).to.be.true;
  });

  it('Should render given albums', () => {
    const wrapper = shallow(<AlbumList albums={mockedAlbums} />);
    const cardsProps = wrapper.find(Card).map(card => card.props());

    expect(mockedAlbums.every(({ title, thumb }, i) =>
      cardsProps[i].header === title && cardsProps[i].image === thumb
    )).to.be.true;
  });

  it('Test albumInfoSearch method', () => {
    const fakeHistory = { push: () => {} };
    const historySpy = sinon.spy(fakeHistory, 'push');
    const albumInfoSearch = sinon.spy();
    const wrapper = shallow(
      <AlbumList history={fakeHistory} albumInfoSearch={albumInfoSearch} />
    );
    const instance = wrapper.instance();

    instance.albumInfoSearch('ALBUM_ID', 'RELEASE_TYPE');
    expect(albumInfoSearch.calledOnceWithExactly('ALBUM_ID', 'RELEASE_TYPE')).to.be.true;
    expect(historySpy.calledOnceWith('/album/ALBUM_ID')).to.be.true;
  })
});
