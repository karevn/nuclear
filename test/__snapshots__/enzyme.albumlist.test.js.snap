// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumList /> Should render given albums 1`] = `
<div
  className="album_list_container"
>
  <div
    className="album_list_cards"
  >
    <Card
      header="TITLE1"
      image="THUMB1"
      key="0"
      onClick={[Function]}
    />
    <Card
      header="TITLE2"
      image="THUMB2"
      key="1"
      onClick={[Function]}
    />
    <Card
      header="TITLE3"
      image="THUMB3"
      key="2"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`<AlbumList /> Should render loader when there is no album 1`] = `
<div
  className="album_list_container"
>
  <DimmerDimmable>
    <Dimmer
      active={true}
    >
      <Loader />
    </Dimmer>
  </DimmerDimmable>
</div>
`;

exports[`<AlbumList /> Test albumInfoSearch method 1`] = `
<div
  className="album_list_container"
>
  <DimmerDimmable>
    <Dimmer
      active={true}
    >
      <Loader />
    </Dimmer>
  </DimmerDimmable>
</div>
`;
